src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -425,9 +434,13 @@ void tryToConnect(NatsUri cur, NatsUri resolved, long now) {
 
             // Make sure the reader and writer are stopped
             long timeoutNanos = timeCheck(end, "waiting for reader");
-            this.reader.stop().get(timeoutNanos, TimeUnit.NANOSECONDS);
+            if (reader.isRunning()) {
+                this.reader.stop().get(timeoutNanos, TimeUnit.NANOSECONDS);
+            }
             timeoutNanos = timeCheck(end, "waiting for writer");
-            this.writer.stop().get(timeoutNanos, TimeUnit.NANOSECONDS);
+            if (writer.isRunning()) {
+                this.writer.stop().get(timeoutNanos, TimeUnit.NANOSECONDS);
+            }

```

Comment:
sometimes stop times out even when it's not running. This guard prevents that since it's already been stopped.